### ISSUE TITLE
Standardising some of the synonym types

### ISSUE THREAD

**Issue by matentzn** [information-artifact-ontology/ontology-metadata#70]
Typical ones we use are:

- layperson
- acronyms
- abbreviation

We may also need things like:

- machine-translated (google translate etc).

AFAIK, there is this problem with OBO format that it translated these:

```
synonym: "IGHD" EXACT ABBREVIATION [DOID:0060870]
```

Into:

```
<owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/MONDO_0000050"/>
        <owl:annotatedProperty rdf:resource="http://www.geneontology.org/formats/oboInOwl#hasExactSynonym"/>
        <owl:annotatedTarget rdf:datatype="http://www.w3.org/2001/XMLSchema#string">IGHD</owl:annotatedTarget>
        <oboInOwl:hasDbXref rdf:datatype="http://www.w3.org/2001/XMLSchema#string">DOID:0060870</oboInOwl:hasDbXref>
        <oboInOwl:hasSynonymType rdf:resource="http://purl.obolibrary.org/obo/mondo#ABBREVIATION"/>
    </owl:Axiom>
```

This is very unfortunate, as it creates incompatible properties all across the OBO sphere.. What is least painful strategy to get out of this problem?

**Comment by matentzn** [comment]
One option would be: 
- we standardise on OWL level (forget about OBO format), and use whatever property we deem fit
- for ontologies that are managed in OBO format, we introduce a sparql transform to map these
- ... until someone changes the mapping in the OBO format itself

**Comment by matentzn** [comment]
For both layperson and akronym, I would suggest new properties in OMO - cant find anything standard out there.

**Comment by cthoyt** [comment]
On slack, I suggested "colloquial" could also be interchangeable with "layperson"

**Comment by cthoyt** [comment]
Btw I'm a huge supporter of standardizing the synonym type definitions across OBO, or at least asking people to make requests to a controlled vocabulary before minting their own

**Comment by cmungall** [comment]
I don't think obo format is a factor

this roundtrips:

```yaml
[Term]
id: X:1
synonym: "foo1" EXACT http://example.org/ABBREVIATION []
synonym: "foo2" EXACT OMO:1234 []
```

and generates:

```xml
    <owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/X_1"/>
        <owl:annotatedProperty rdf:resource="http://www.geneontology.org/formats/oboInOwl#hasExactSynonym"/>
        <owl:annotatedTarget rdf:datatype="http://www.w3.org/2001/XMLSchema#string">foo1</owl:annotatedTarget>
        <oboInOwl:hasSynonymType rdf:resource="http://example.org/ABBREVIATION"/>
    </owl:Axiom>
    <owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/X_1"/>
        <owl:annotatedProperty rdf:resource="http://www.geneontology.org/formats/oboInOwl#hasExactSynonym"/>
        <owl:annotatedTarget rdf:datatype="http://www.w3.org/2001/XMLSchema#string">foo2</owl:annotatedTarget>
        <oboInOwl:hasSynonymType rdf:resource="http://purl.obolibrary.org/obo/OMO_1234"/>
    </owl:Axiom>
```

the one constraint is that we can't use a non-OBO CURIE as the type without modifying the owlapi code and coordinating those changes.

The use of a plain uppercase string is just convention. I think if we come up with a standard for this I can commit to GO, Uberon, Mondo, CL, etc all moving. (I need to check about the syngo preferred label though)

A minor inconvenience is that there is a special translation of synonym type definitions

```yaml
synonymtypedef: http://example.org/ABBREVIATION "full URI test"
synonymtypedef: OMO:1234 "CURIE test" EXACT

[Term]
id: X:1
synonym: "foo1" EXACT http://example.org/ABBREVIATION []
synonym: "foo2" EXACT OMO:1234 []
```

these get modeled as annotation properties:

```xml
    <!-- http://example.org/ABBREVIATION -->

    <owl:AnnotationProperty rdf:about="http://example.org/ABBREVIATION">
        <rdfs:label rdf:datatype="http://www.w3.org/2001/XMLSchema#string">full URI test</rdfs:label>
        <rdfs:subPropertyOf rdf:resource="http://www.geneontology.org/formats/oboInOwl#SynonymTypeProperty"/>
    </owl:AnnotationProperty>



    <!-- http://purl.obolibrary.org/obo/OMO_1234 -->

    <owl:AnnotationProperty rdf:about="http://purl.obolibrary.org/obo/OMO_1234">
        <oboInOwl:hasScope rdf:resource="http://www.geneontology.org/formats/oboInOwl#hasExactSynonym"/>
        <rdfs:label rdf:datatype="http://www.w3.org/2001/XMLSchema#string">CURIE test</rdfs:label>
        <rdfs:subPropertyOf rdf:resource="http://www.geneontology.org/formats/oboInOwl#SynonymTypeProperty"/>
    </owl:AnnotationProperty>
```

That was a decision we took in the initial obo2owl mapping as at that time populating your ontology with individuals had devastating effects on reasoning. I am not sure it was the correct choice, but this would be expensive to undo.

I propose:

 1. OMO IDs/URIs are minted for synonym types
 2. These are modeled as annotation properties subproperties of SynonymTypeProperty
 3. We have linkml/shacl checks to ensure they are used in structurally the correct way
 4. Ontologies have a 6 month timeframe to adopt

Although it should not be coupled to this ticket, we need to document use cases for synonym scopes and types better

**Comment by matentzn** [comment]
I LOVE IT THANK YOU!

**Comment by shawntanzk** [comment]
We are also keen on having abbreviation and acronym in CL and UBERON - see https://github.com/obophenotype/cell-ontology/issues/1441

**Comment by bvarner-ebi** [comment]
Is a separate ticket required to bring the abbreviation and acronym synonym types into CL and UBERON? Kindly advise what is required to facilitate this. @shawntanzk, given your technical expertise, would you be able to help with the implementation plan?

**Comment by shawntanzk** [comment]
> given your technical expertise
lol 100% no expertise >.< 

I think this requires community agreement and support/need to push this up the priority list more than anything

**Comment by cthoyt** [comment]
@matentzn I think this is done

**Comment by matentzn** [comment]
Closed as per #159
